src/EditorFeatures/Core/GoToDefinition/AbstractGoToDefinitionService.cs

```diff
@@ -42,67 +42,75 @@ internal abstract class AbstractGoToDefinitionService : AbstractFindDefinitionSe
         async Task<IEnumerable<INavigableItem>?> IGoToDefinitionService.FindDefinitionsAsync(Document document, int position, CancellationToken cancellationToken)
             => await FindDefinitionsAsync(document, position, cancellationToken).ConfigureAwait(false);
 
-        private bool TryNavigateToSpan(Document document, int position, CancellationToken cancellationToken)
+        private static Task<INavigableLocation?> GetNavigableLocationAsync(
+            Document document, int position, CancellationToken cancellationToken)
         {
             var solution = document.Project.Solution;
             var workspace = solution.Workspace;
             var service = workspace.Services.GetRequiredService<IDocumentNavigationService>();
 
-            var options = new NavigationOptions(PreferProvisionalTab: true, ActivateTab: true);
-            return _threadingContext.JoinableTaskFactory.Run(() =>
-                service.TryNavigateToPositionAsync(workspace, document.Id, position, virtualSpace: 0, options, cancellationToken));
+            return service.GetLocationForPositionAsync(
+                workspace, document.Id, position, virtualSpace: 0,
+                new NavigationOptions(PreferProvisionalTab: true, ActivateTab: true), cancellationToken);
         }
 
         public bool TryGoToDefinition(Document document, int position, CancellationToken cancellationToken)
+            => throw new NotImplementedException("Use FindDefinitionLocationAsync instead.");
```

Comment:
Why is this method here at all at this point?

Reply:
removed